opensearch-operator/pkg/builders/cluster.go

```diff
@@ -972,3 +984,80 @@ func DataNodesCount(ctx context.Context, k8sClient client.Client, cr *opsterv1.O
 	}
 	return count
 }
+
+func NewServiceMonitor(cr *opsterv1.OpenSearchCluster) *v1.ServiceMonitor {
+
+	labels := map[string]string{
+		ClusterLabel: cr.Name,
+	}
+	selector := metav1.LabelSelector{
+		MatchLabels: labels,
+	}
+
+	namespaceSelector := v1.NamespaceSelector{
+		Any:        false,
+		MatchNames: []string{cr.Namespace},
+	}
+
+	if cr.Spec.General.Monitoring.ScrapInterval == "" {
+		cr.Spec.General.Monitoring.ScrapInterval = "30s"
+	}
+	user := v1.BasicAuth{}
+
+	monitorUser := cr.Spec.General.Monitoring.MonitoringUserSecret
+	if monitorUser == "" {
+		monitorUser = "admin"
+		user = v1.BasicAuth{
+			Username: corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: cr.Name + monitorUser + "-password"},
+				Key:                  "username",
+			},
+			Password: corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: cr.Name + monitorUser + "-password"},
+				Key:                  "password",
+			},
+		}
+	} else {
+		user = v1.BasicAuth{
+			Username: corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: cr.Spec.General.Monitoring.MonitoringUserSecret},
+				Key:                  "username",
+			},
+			Password: corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{Name: cr.Spec.General.Monitoring.MonitoringUserSecret},
+				Key:                  "password",
+			},
+		}
+
+	}
+
+	return &v1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + "-monitor",
+			Namespace: cr.Namespace,
+			Labels:    labels,
+		},
+		Spec: v1.ServiceMonitorSpec{
+			JobLabel: cr.Name + "-monitor",
+			TargetLabels: []string{
+				cr.Name,
+			},
+			PodTargetLabels: []string{
+				cr.Name,
+			},
+			Endpoints: []v1.Endpoint{
+				{Port: "9200",
```

Comment:
Port cannot be hardcoded as it is configurable in the custom resource (´general.httpPort´).

Reply:
fixed